mycore-base/src/main/java/org/mycore/services/queuedjob/MCRJobQueue.java

```diff
@@ -349,6 +349,44 @@ private MCRJob getJob(Class<? extends MCRJobAction> action, Map<String, String>
         }
     }
 
+    /**
+     * returns specific jobs by the given parameters or an empty list.
+     *
+     * @param params the parameters
+     * @return the job
+     */
+    public List<MCRJob> getJobs(Map<String, String> params) {
+        return getJobs(action, params);
+    }
+
+    private List<MCRJob> getJobs(Class<? extends MCRJobAction> action, Map<String, String> params) {
+        if (!running) {
+            return null;
+        }
+
+        EntityManager em = MCREntityManagerProvider.getCurrentEntityManager();
+
+        StringBuilder qStr = new StringBuilder("FROM MCRJob job WHERE action = '"
+                + action.getName() + "' ");
+        for (String paramKey : params.keySet()) {
+            qStr.append(" AND job.parameters['")
+                    .append(paramKey)
+                    .append("'] = '")
+                    .append(params.get(paramKey))
+                    .append('\'');
```

Comment:
The current state is vulnerable to sql injection. 
Also the code ist 99% duplicate. 
The method getJob is private so you can change it to
private <T> T getJob(Class<? extends MCRJobAction> action, Map<String, String> params, Function<TypedQuery<MCRJob>, T> queryTransfomer)  and reuse it.

Please also use the query api to prevent SQL Injection.
For examples see:
org.mycore.pi.MCRPIManager